syntax = "proto3";

package rins;

enum MessageID {
  ID_UNKNOWN = 0;
  ID_GetRouteRequest = 1;
  ID_GetRouteResponse = 2;
  ID_ReportStatusRequest = 3;
}

message HostInfo {
  int32 ip = 1;
  int32 port = 2;
}

message GetRouteRequest {
  int32 modid = 1;
  int32 cmdid = 2;
}

message GetRouteResponse {
  int32 modid = 1;
  int32 cmdid = 2;
  repeated HostInfo host = 3;
}

message HostCallResult {
  int32 ip = 1;
  int32 port = 2;
  uint32 succ = 3;
  uint32 err = 4;
  bool overload = 5;
}

message ReportStatusRequest {
  int32 modid = 1;
  int32 cmdid = 2;
  int32 caller = 3;
  repeated HostCallResult results = 4;
  uint32 timestamp = 5;
}